user-guide/Troubleshooting/Procedures/SLScripting_FinalizerException.md

```diff
@@ -0,0 +1,259 @@
+--------------------------------------
+Module:      0ce01a04
+Assembly:    ExampleProtocol.1.0.0.1.QAction.1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
+Token:       02000002
+MethodTable: 0ce02f2c
+EEClass:     0da35d64
+Name:        Skyline.Protocol.MyExtension.MyClass
+--------------------------------------
+Module:      0ce01dbc
+Assembly:    ExampleProtocol.1.0.0.1.QAction.900012, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
+--------------------------------------
+Module:      0ce021e8
+Assembly:    ExampleProtocol.1.0.0.1.QAction.2, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
+--------------------------------------
+```
+
+The output lists all the assemblies and for each assembly that defines this type the type is mentioned. In this case, we see that the assembly `ExampleProtocol.1.0.0.1.QAction.1` defines a type `Skyline.Protocol.MyExtension.MyClass` so we should inspect the Finalizer of the `MyClass` type defined in QAction 1 of the `ExampleProtocol` connector version `1.0.0.1`.
+
+In WinDbg, you can use the `!dumpclass` command to retrieve more information about the type using its EEClass value as argument: `!dumpclass 0da35d64`:
+
+```txt
+0:003> !dumpclass 0da35d64
+Class Name:      Skyline.Protocol.MyExtension.MyClass
+mdToken:         02000002
+File:            ExampleProtocol.1.0.0.1.QAction.1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
+Parent Class:    72dc6118
+Module:          0ce01a04
+Method Table:    0ce02f2c
+Vtable Slots:    4
+Total Method Slots:  5
+Class Attributes:    100001  
+Transparency:        Critical
+NumInstanceFields:   0
+NumStaticFields:     0
+```
+
+Use the `!dumpmt -md 0ce02f2c` (where 0ce02f2c is the Method Table address of the type as seen from the output of the `!dumpclass` command) to see the methods this type defines:
+
+```txt
+0:003> !dumpmt -md 0ce02f2c
+EEClass:         0da35d64
+Module:          0ce01a04
+Name:            Skyline.Protocol.MyExtension.MyClass
+mdToken:         02000002
+File:            ExampleProtocol.1.0.0.1.QAction.1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
+BaseSize:        0xc
+ComponentSize:   0x0
+Slots in VTable: 6
+Number of IFaces in IFaceMap: 0
+--------------------------------------
+MethodDesc Table
+   Entry MethodDe    JIT Name
+731a9a58 72dc9e9c PreJIT System.Object.ToString()
+7326a370 72f261d8 PreJIT System.Object.Equals(System.Object)
+73198e60 72f261f8 PreJIT System.Object.GetHashCode()
+0ce20840 0ce02f08    JIT Skyline.Protocol.MyExtension.MyClass.Finalize()
+0d37de08 0ce02f18   NONE Skyline.Protocol.MyExtension.MyClass..ctor()
+0ce20820 0ce02efc    JIT Skyline.Protocol.MyExtension.MyClass.PerformAction()
+```
+
+The see the IL code of the Finalize method, use the `!u 0ce02f08` command (where 0ce02f08 is the address denoted for the Finalize method in the output of the `!dumpmt` command).
+
+### Using the eestack command
```

Comment:
As the first sentence below this suggests that this is an alternative to what was described above, could we add another subtitle for the section above to make the structure of the page more clear?

Reply:
Restructured it so that it is more clear.